packages/experimental/TabList/src/Tab/useTabAnimation.ts

```diff
@@ -28,6 +29,11 @@ export function useTabAnimation(
   const { addTabLayout, selectedKey, layout, updateAnimatedIndicatorStyles, vertical } = context;
   const { tabKey } = props;
 
+  // This tab's layout rect from the onLayout callback
+  const [rect, setRect] = React.useState<LayoutRectangle>();
+  // Whenever we get new layout info, we need to re-update the global tab layout map in our context. This flag triggers the logic in the useEffect hook below.
+  const [shouldUpdateLayoutInfo, setShouldUpdateLayoutInfo] = React.useState(false);
```

Comment:
the useEffect hook will already trigger based on the dependencies; what are we gaining from adding a separate flag that limits the effect to only triggering against `[rect, setRect]`?  If we want to limit updates to processing after the `onTabLayout` event, should we move this code into the callback and get rid of the `useEffect` hook?

Reply:
The issue is that for non-win32 platforms, it's not guaranteed that we'll have the TabList's layout information whenever the Tab's layout callback runs. If we want to check whether Tab's the layout information "makes sense" by comparing it to the TabList's layout information, we need to wait until we have both the Tab and TabList layouts, which is what the useEffect is for. 

This could also be solved by forking the hook because non-win32 platforms don't seem to have the same scrollview layout issues that win32 has which put those checks there in the first place.

Reply:
My top-line thinking is around:
- if `setShouldUpdateLayoutInfo(true)` is only invoked after `setRect(e.nativeEvent.layout)`, is the `rect &&` check not sufficient?

though maybe your point is less about first-run and more about dynamic runtime changes to other props (e.g. `tokens.indicatorMargins`) where while reliant on those properties, the layout results we're using are dependent on them as well, so the dynamic run of the useEffect hook based on (e.g. `tokens.indicatorMargins`) wouldn't be valid because we de-synced our layout inputs/outputs?  In which case, maybe after using the `rect` layout results we could toss them (`setRect(null)`) and that would de-dupe our state-tracking of "is `rect` up to date?" ?